### PR TITLE
Quick fix for matches not display properly in the report viewer.

### PR DESCRIPTION
**Pull request by sebinside**
This fix shall quickly solve the match display in the report viewer. It does not completely solve the underlying issue, see #686. 

In addition, the serialization of clusters is disabled in the report viewer output. 

Last, this PR also adds a very simple version checking in the report viewer.

### CODE REVIEW

File: core/src/main/java/de/jplag/reporting/reportobject/ReportObjectFactory.java [hunk 1]
```diff
@@ -42,6 +43,9 @@ public class ReportObjectFactory {
     private static final ToDiskWriter fileWriter = new ToDiskWriter();
     public static final String OVERVIEW_FILE_NAME = "overview.json";
     public static final String SUBMISSIONS_FOLDER = "submissions";
+
+    // This shall be moved to a better visible and upgradable position. Shall be fixed in a future version.
```
Comment (JanWittler) [inline core/src/main/java/de/jplag/reporting/reportobject/ReportObjectFactory.java:47]:
```suggestion
    // TODO: This shall be moved to a better visible and upgradable position. Shall be fixed in a future version.
```

File: core/src/main/java/de/jplag/reporting/reportobject/mapper/ClusteringResultMapper.java [hunk 2]
```diff
@@ -21,7 +19,11 @@ public ClusteringResultMapper(Function<Submission, String> submissionToIdFunctio
 
     public List<Cluster> map(JPlagResult result) {
         var clusteringResult = result.getClusteringResult();
-        return clusteringResult.stream().map(ClusteringResult::getClusters).flatMap(Collection::stream).map(this::convertCluster).toList();
+
+        // Clustering disabled for the report viewer. Shall be fixed in a future version.
```
Comment (JanWittler) [inline core/src/main/java/de/jplag/reporting/reportobject/mapper/ClusteringResultMapper.java:23]:
```suggestion
        // TODO: Clustering disabled for the report viewer. Shall be fixed in a future version.
```

File: core/src/main/java/de/jplag/reporting/reportobject/model/Version.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * This record represents a version number according to <a href="https://semver.org/">https://semver.org/</a>. This
+ * version number is used to compare generated result.zip files with the version of the report-viewer.
+ * @param major MAJOR version when you make incompatible API changes
+ * @param minor MINOR version when you add functionality in a backwards compatible manner
+ * @param patch PATCH version when you make backwards compatible bug fixes
+ */
+public record Version(@JsonProperty("major") int major, @JsonProperty("minor") int minor, @JsonProperty("patch") int patch) {
+}
```
Comment (JanWittler) [inline core/src/main/java/de/jplag/reporting/reportobject/model/Version.java:13]: Maybe a bit over engineered, don't you think? 😄 I'd say a version String would be sufficient.

File: report-viewer/src/model/factories/OverviewFactory.ts [hunk 4]
```diff
@@ -68,6 +81,14 @@ export class OverviewFactory {
...
   }
 
+  static compareVersions(jsonVersion: Version, reportViewerVersion: Version) {
+    if(jsonVersion.major !== reportViewerVersion.major ||
+       jsonVersion.minor !== reportViewerVersion.minor ||
+       jsonVersion.patch !== reportViewerVersion.patch) {
+        console.warn("The result's version tag does not fit the report viewer's version. Trying to read it anyhow but be careful.")
+       }
+  }
+
```
Comment (JanWittler) [inline report-viewer/src/model/factories/OverviewFactory.ts:91]: If we simplify it to String, we can use `semver` to compare the version here

https://stackoverflow.com/questions/6832596/how-can-i-compare-software-version-number-using-javascript-only-numbers

File: report-viewer/vue.config.js [hunk 5]
```diff
@@ -11,5 +11,6 @@ module.exports = defineConfig({
         chunks: "all",
       },
     },
+    devtool: 'source-map'
```
Comment (JanWittler) [inline report-viewer/vue.config.js:14]: What is this for?

File: core/src/main/java/de/jplag/reporting/reportobject/model/OverviewReport.java [hunk 6]
```diff
@@ -8,6 +8,8 @@
 
 public record OverviewReport(
 
+        @JsonProperty("version") Version version,
```
Comment (JanWittler) [inline core/src/main/java/de/jplag/reporting/reportobject/model/OverviewReport.java:11]: Maybe we name this property rather `JPlag-version` or similar? I could image in the future to introduce an actual versioning of the JSON report structure, and then we would have two version numbers: version of the json format and the version of JPlag used to generate it.

**Review by JanWittler — CHANGES_REQUESTED** [review]

File: report-viewer/src/model/factories/ComparisonFactory.ts [hunk 7]
```diff
@@ -21,6 +21,13 @@ export class ComparisonFactory {
     );
 
     const matches = json.matches as Array<Record<string, unknown>>;
+    
+    // Quick fix for issue #658, requires deeper insights to be resolved completely
```
[the remark this answers is not in the document]
Comment (sebinside) [reply report-viewer/src/model/factories/ComparisonFactory.ts:25]: Unfortunately, SonarCloud does not like correct annotations in comments ;)

[the remark this answers is not in the document]
Comment (JanWittler) [reply report-viewer/src/model/factories/ComparisonFactory.ts:25]: We can still merge it even if SonarCloud is complaining.

File: report-viewer/src/model/factories/OverviewFactory.ts (hunk 4 above)
Comment (sebinside) [reply report-viewer/src/model/factories/OverviewFactory.ts:91]: I don't think that replacing a quite simple solution with yet another dependency is a that good approach :D

File: report-viewer/vue.config.js (hunk 5 above)
Comment (sebinside) [reply report-viewer/vue.config.js:14]: Used for debugging the report viewer. I also wondered why it was not already there.

**Review by JanWittler — APPROVED** [review]

**Pull request merged**